### DISCUSSION TITLE
How to download csv file?

### DISCUSSION THREAD

**Discussion by ginnyvt** [nuxt/nuxt#16740] [Questions]
Hi guys!

I have a server API that is used to download a CSV file (blob format). I wonder how to implement the fetch in the client side?

**Comment by sevillaarvin** [comment]
Can you clarify what you mean? Downloading something is just a GET request.

If you want to process the csv in your app perhaps `useFetch("/path/to/my/file.csv")`

If you want the user to be able to download it from the browser to their computer, use `<a href="/path/to/my/file.csv">Download</a>`

**Reply by ginnyvt** [reply]
I have tried similar approach using the `<a href="/path/to/my/file.csv" download="my-file.csv">Download</a>` , but I cannot get the downloaded file with the modified name and csv extension. 

I also tried using fetch() but receive the data as `Blob {}`.

**Reply by sevillaarvin** [reply]
Try this:

```vue
<script setup>
if(process.client) {
  $fetch("/test.csv", { responseType: 'blob' })
    .then((content) => {
      const a = document.createElement("a");
      const url = window.URL.createObjectURL(content);
      a.href = url;
      a.download = "changed_name.csv";
      a.click();
    })
}
</script>
```

References:

https://stackoverflow.com/a/56195387
